app/src/main/java/fr/free/nrw/commons/upload/worker/WorkRequestHelper.kt

```diff
@@ -11,7 +12,16 @@ import java.util.concurrent.TimeUnit
 class WorkRequestHelper {
 
     companion object {
+        @Volatile
+        private var isUploadWorkerRunning = false
+
         fun makeOneTimeWorkRequest(context: Context, existingWorkPolicy: ExistingWorkPolicy) {
+
+            if (isUploadWorkerRunning) {
```

Comment:
explain this design choice, for instance "we only allow one upload at a time because upload is supposed to work as a queue. Also, parallel uploads were triggering errors and did not improve overall throughput".

Reply:
Could this be related to as isUploadWorkerRunning is volatile which could lead to race condition between different threads?
https://www.baeldung.com/java-volatile-vs-atomic#synchronization-problem

Reply:
How about making isUploadWorkerRunning static instead of volatile?
And checking+updating it in a synchronized block, something like this:

(class level)
private static final Object lock = new Object(); // Lock object

(inside the method)
synchronized (lock) { // Acquire the lock
                if (isUploadWorkerRunning) {
                    Timber.e("UploadWorker is already running");
                    return;
                } else {
                    Timber.e("Setting isUploadWorkerRunning to true");
                    isUploadWorkerRunning = true;
                }
}
... proceed with the upload ...

Reply:
Sure, I will share a newer commit for it!

Reply:
I was testing today and somehow was able to reproduce the issue with multiple uploads happening at same time


https://github.com/user-attachments/assets/c3db0562-ff4f-4578-bd6b-9d3a98cf8d1a



https://github.com/user-attachments/assets/d9a56ad4-331e-4598-a4f1-dcf67f0d9cb1